app/views/admin/shared/_tooltip.html.haml

```diff
@@ -1,5 +1,8 @@
-%div{"data-controller": "tooltip"}
-  %a{"data-tooltip-target": "element", href: link, class: link_class}
+- tooltip_placement = defined?(placement) ? placement : "top"
+- tooltip_link = defined?(link) ? link : ""
+- tooltip_link_class = defined?(link_class) ? link_class : ""
+%div{"data-controller": "tooltip", "data-tooltip-placement-value": tooltip_placement }
+  %a{"data-tooltip-target": "element", href: tooltip_link, class: tooltip_link_class}
```

Comment:
I'm yet to find a prettier Ruby pattern for handling optional undefined variables in views..

Reply:
I quite like to declare local variables at the top:

```haml
- link ||= ""
- link_class ||= ""
```

The alternative would be
```haml
- tooltip_link = local_assigns[:link].to_s
```

or

```haml
  %a{"data-tooltip-target": "element", href: local_assigns[:link], class: local_assigns[:link_class]}
```